budgetportal/tests/test_infrastructure_projects.py

```diff
@@ -0,0 +1,100 @@
+from django.test import TestCase
+
+from budgetportal.models import InfrastructureProject
+
+
+class ProjectedExpenditureTestCase(TestCase):
+    """ Unit tests for get_projected_expenditure function """
+
+    def setUp(self):
+        self.fake_records = [
+            {'Financial Year': 2016, 'Amount': 100},
+            {'Financial Year': 2017, 'Amount': 100},
+            {'Financial Year': 2018, 'Amount': 100},
+            {'Financial Year': 2019, 'Amount': 100},
+            {'Financial Year': 2020, 'Amount': 100},
+            {'Financial Year': 2021, 'Amount': 100}
+        ]
+        self.project = InfrastructureProject()
+
+    def test_success(self):
+        self.project.records = self.fake_records
+        projected_expenditure = self.project.get_projected_expenditure()
+        self.assertEqual(projected_expenditure, 300)
+
+    def test_empty_records_returns_zero(self):
+        self.project.records = []
+        projected_expenditure = self.project.get_projected_expenditure()
+        self.assertEqual(projected_expenditure, 0)
+
+
+class CleanedCoordinatesTestCase(TestCase):
+    """ Unit tests for get_cleaned_coordinates function """
+
+    def setUp(self):
+        self.project = InfrastructureProject()
+
+    def test_standard_format(self):
+        self.project.gps_codes = '-26.378582,27.654933'
```

Comment:
like I mentioned above, if you factor out the coordinate parsing to one or two functions, you can test those directly as functional units without needing to instantiate InfrastructureProject() each time